[docs] Update the readme (#2)
wentwrong/testcafe-plugin: README.md

# Jenkins TestСafe Plugin

[![GitHub release](https://img.shields.io/github/v/release/wentwrong/testcafe-jenkins-plugin?include_prereleases)](https://github.com/wentwrong/testcafe-jenkins-plugin/releases)

This plugin attaches TestCafe artifacts (screenshots and videos) to the Jenkins report and displays them on the build page.

![Jenkins TestCafe Plugin](media/jenkins-result.png)

## Usage

> > See [Integrate TestCafe with Jenkins](https://devexpress.github.io/testcafe/documentation/guides/continuous-integration/jenkins.html) for information on how to configure Jenkins initially to run TestCafe.

Reviewer: ```suggestion
> See the following topic for information on how to configure Jenkins to run TestCafe:  [Integrate TestCafe with Jenkins](https://devexpress.github.io/testcafe/documentation/guides/continuous-integration/jenkins.html).
```

This plugin requires that you install and use the [TestCafe Jenkins reporter](https://www.npmjs.com/package/testcafe-reporter-jenkins):

```sh
npm i testcafe-reporter-jenkins
```

> You can add `testcafe-reporter-jenkins` to a step that installs TestCafe:

Reviewer: ```suggestion
You can add `testcafe-reporter-jenkins` during the step that installs TestCafe:
```

![Install the TestCafe Jenkins reporter](media/jenkins-install-reporter.png)

In the command that runs TestCafe, use the [-r flag](https://devexpress.github.io/testcafe/documentation/reference/command-line-interface.html#-r-nameoutput---reporter-nameoutput) to specify the Jenkins reporter and the report file name:

```sh
testcafe chrome e2e/**/* -r jenkins:report.xml
```

### Use in Freestyle Jobs

> Add a post-build action **Publish JUnit test result report**. In **Additional test report features**, click **Add** and select **Include links to TestCafe artifacts**:

Reviewer: ```suggestion
Add the **Publish JUnit test result report** post-build action. In **Additional test report features**, click **Add** and select **Include links to TestCafe artifacts**:
```

![Use TestCafe in a Freestyle Job](media/jenkins-configure-plugin.png)

Ensure that **Retain long standard output/error** is enabled so that Jenkins does not truncate the screenshot and video data.

### Use in Pipelines

In the [Jenkinsfile](https://www.jenkins.io/doc/book/pipeline/jenkinsfile/), add a [post-build step](https://www.jenkins.io/doc/book/pipeline/syntax/#post) that publishes the JUnit report:

```
post {
    always {
        junit keepLongStdio: true,
              testDataPublishers: [[$class: 'TestCafePublisher']],
              testResults: 'report.xml'
    }
}
```

> > Notice that the `keepLongStdio` parameter must be set to `true`, otherwise the JUnit truncates the test's output.

Reviewer: ```suggestion
> Notice that the `keepLongStdio` parameter must be set to `true`, otherwise the JUnit truncates the test output.
```

> Visit [this repository](https://github.com/wentwrong/experiments-with-jenkins) for an example of how to run tests distributed among multiple agents in parallel.

Reviewer: ```suggestion
To see an example of how to run tests distributed among multiple agents in parallel, refer to the following GitHub repository: [Experiments with Jenkins](https://github.com/wentwrong/experiments-with-jenkins).
```

## Troubleshooting

**The video player is blank or the video does not start**

> This issue can occur due to the [Content Security Policy](https://wiki.jenkins.io/display/JENKINS/Configuring+Content+Security+Policy) (CSP) in Jenkins. Check the browser console to see what policies are violated.

Reviewer: ```suggestion
This issue can occur due to the [Content Security Policy](https://wiki.jenkins.io/display/JENKINS/Configuring+Content+Security+Policy) (CSP) in Jenkins. Check the browser console to see which policies are violated.
```
